x/evm/types/state_object.go

```diff
@@ -77,20 +78,33 @@ type (
 	}
 )
 
-func newObject(db *CommitStateDB, accProto authexported.Account) *stateObject {
-	acc, ok := accProto.(*types.Account)
+func newStateObject(db *CommitStateDB, accProto authexported.Account) *stateObject {
+	var (
+		ethermintAccount *types.Account
+		ok               bool
+	)
+	ethermintAccount, ok = accProto.(*types.Account)
 	if !ok {
-		panic(fmt.Sprintf("invalid account type for state object: %T", accProto))
+		// check if account is a BaseAccount
+		baseAcc, ok := accProto.(*auth.BaseAccount)
+		if !ok {
+			panic(fmt.Sprintf("invalid account type for state object: %T", accProto))
+		}
+		ethermintAccount = &types.Account{
+			BaseAccount: baseAcc,
+			CodeHash:    emptyCodeHash,
+		}
```

Comment:
I actually used to have this and it caused many issues and errors when converting the account type where it would drop certain values, which is why later I took it out and only supported the one type. I can try to revisit but be very careful with this because it caused me a lot of headachep previously

Reply:
updated. Using the ethermint types `ProtoAccount` solved the issue